go/vt/wrangler/vexec.go

```diff
@@ -499,11 +528,41 @@ func (wr *Wrangler) getStreams(ctx context.Context, workflow, keyspace string) (
 			sourceShards.Insert(status.Bls.Shard)
 			rsrStatus = append(rsrStatus, status)
 
+			if status.Message == workflow2.Frozen {
+				rsr.Frozen = true
+			}
+
+			// MaxVReplicationLag is the time since the last event was processed from the source
+			// The last event can be an actual binlog event or a heartbeat in case no binlog events occur within (default) 1 second
 			timeUpdated := time.Unix(status.TimeUpdated, 0)
 			replicationLag := time.Since(timeUpdated)
 			if replicationLag.Seconds() > float64(rsr.MaxVReplicationLag) {
 				rsr.MaxVReplicationLag = int64(replicationLag.Seconds())
 			}
+
+			// MaxVReplicationTransactionLag estimates the actual lag between the source and the target
+			// If we are still processing source events it is the difference b/w current time and the timestamp of the last event
+			// If heartbeats are more recent than the last event, then the lag is the time since the last heartbeat as
+			// there can be an actual event immediately after the heartbeat, but which has not yet
+			// been processed on the target
+			// We don't allow switching during the copy phase, so in that case we just return a large lag.
+			// All timestamps are in seconds since epoch
+			lastTransactionTimestamp := status.TransactionTimestamp
+			lastHeartbeatTime := status.TimeHeartbeat
+			if status.State != "Copying" {
```

Comment:
IMO it's easier to read if the predicate is close to the action. In this case what to do when State == Copying is much later. i.e. I think it's easier to follow if we did:
```
if status.State == "Copying" {
...
} else {
}
```

I'm guilty of doing it the current way though too.

Reply:
agree, fixed